### PR TITLE
Refactor `setup.py` and solve tests deprecation warning

### PR DESCRIPTION
**Pull request by uhlajs**
Add some small modification of dependencies handling base on [OpenAI Gym](https://github.com/openai/gym). Also, solve the deprecation issue with `pytest_cases` and add scrip for record environment.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,8 @@ python:
 
 # command to install dependencies
 install:
-  - pip install -e .
+  - pip install -e '.[all]'
+  - pip freeze
```
Comment (koulanurag) [inline .travis.yml:8]: remove `freeze` line

**Review by koulanurag — COMMENTED** [review]
All good!

File: .travis.yml (hunk 1 above)
Comment (uhlajs) [reply .travis.yml:8]: Is there any reason for that? With this line, you can see the exact version of packages and easily debug the potential issue in Travis CI.

Comment (koulanurag) [reply .travis.yml]: Though I l usually like to keep things to minimal. 

As you suggested, I feel this line wouldn't hurt. So, I will restore it back.

**Pull request merged**